src/react-contenteditable.js

```diff
@@ -20,13 +20,25 @@ export default class ContentEditable extends React.Component {
   }
 
   shouldComponentUpdate(nextProps) {
-    return !this.htmlEl || nextProps.html !== this.htmlEl.innerHTML ||
-            this.props.disabled !== nextProps.disabled;
+    // We need not rerender if the change of props simply reflects the user's
+    // edits. Rerendering in this case would make the cursor/caret jump.
+    return (
+      // Rerender if there is no element yet... (somehow?)
+      !this.htmlEl
+      // ...or if html really changed... (programmatically, not by user edit)
+      || ( nextProps.html !== this.htmlEl.innerHTML
+        && nextProps.html !== this.props.html )
+      // ...or if editing is enabled or disabled.
+      || this.props.disabled !== nextProps.disabled
+    );
   }
 
   componentDidUpdate() {
     if ( this.htmlEl && this.props.html !== this.htmlEl.innerHTML ) {
-     this.htmlEl.innerHTML = this.props.html;
+      // Perhaps React's VDOM (which gets outdated because we often prevent
+      // rerendering) thought that updating was not required. So we update the
+      // element manually now.
```

Comment:
Yes I think it's a better comment and explains better the issue I raised here: http://stackoverflow.com/a/27255103/82609